libkbfs/identify_util.go

```diff
@@ -170,48 +180,67 @@ func identifyUID(ctx context.Context, nug normalizedUsernameGetter, identifier i
 	return nil
 }
 
+// identifyUserToChan calls identifyUser and plugs the result into the error channnel.
+func identifyUserToChan(ctx context.Context, nug normalizedUsernameGetter,
+	identifier identifier, name libkb.NormalizedUsername, uid keybase1.UID,
+	isPublic bool, errChan chan error) {
+	errChan <- identifyUser(ctx, nug, identifier, name, uid, isPublic)
+}
+
+// identifyUsers identifies the users in the given maps.
+func identifyUsers(ctx context.Context, nug normalizedUsernameGetter,
+	identifier identifier, users map[keybase1.UID]libkb.NormalizedUsername,
+	public bool) error {
+	eg, ctx := errgroup.WithContext(ctx)
+
+	// TODO: limit the number of concurrent identifies?
+	// TODO: implement a version of errgroup with limited concurrency.
+	for uid, name := range users {
+		uid, name := uid, name
+		eg.Go(func() error {
+			return identifyUser(ctx, nug, identifier, name, uid, public)
+		})
+	}
+
+	return eg.Wait()
+}
+
 // identifyUserList identifies the users in the given list.
 func identifyUserList(ctx context.Context, nug normalizedUsernameGetter, identifier identifier, uids []keybase1.UID, public bool) error {
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
+	eg, ctx := errgroup.WithContext(ctx)
```

Comment:
Add a comment here too about only using this when the usernames aren't known, like in rekeying